In [2]:
import pandas as pd
import numpy as np
import os

In [8]:
bank = pd.read_csv('bank_market.csv.csv')
bank

Cust_num  age           job   marital  education default  balance  \
0             1   58    management   married   tertiary      no     2143   
1             2   44    technician    single  secondary      no       29   
2             3   33  entrepreneur   married  secondary      no        2   
3             4   47   blue-collar   married    unknown      no     1506   
4             5   33       unknown    single    unknown      no        1   
...         ...  ...           ...       ...        ...     ...      ...   
45206     45207   51    technician   married   tertiary      no      825   
45207     45208   71       retired  divorced    primary      no     1729   
45208     45209   72       retired   married  secondary      no     5715   
45209     45210   57   blue-collar   married  secondary      no      668   
45210     45211   37  entrepreneur   married  secondary      no     2971   

      housing loan    contact  day month  duration  campaign  pdays  previous  \
0         yes   no    unknown    5   may       261         1     -1         0   
1         yes   no    unknown    5   may       151         1     -1         0   
2         yes  yes    unknown    5   may        76         1     -1         0   
3         yes   no    unknown    5   may        92         1     -1         0   
4          no   no    unknown    5   may       198         1     -1         0   
...       ...  ...        ...  ...   ...       ...       ...    ...       ...   
45206      no   no   cellular   17   nov       977         3     -1         0   
45207      no   no   cellular   17   nov       456         2     -1         0   
45208      no   no   cellular   17   nov      1127         5    184         3   
45209      no   no  telephone   17   nov       508         4     -1         0   
45210      no   no   cellular   17   nov       361         2    188        11   

      poutcome    y  
0      unknown   no  
1      unknown   no  
2      unknown   no  
3      unknown   no  
4      unknown   no  
...        ...  ...  
45206  unknown  yes  
45207  unknown  yes  
45208  success  yes  
45209  unknown   no  
45210    other   no  

[45211 rows x 18 columns]

In [185]:
#min age of customers 
min_custage = min(bank.age)
min_custage

18

In [187]:
#max customer age
max_custage = max(bank.age)
max_custage

95

In [195]:
#defaulter customer min age
def_cust = bank[bank["default"]== "yes"]
def_cust_min_age = min(def_cust.age)
def_cust_min_age

21

In [197]:
#Defaulter customer max age
def_cust_max_age = max(def_cust.age)
def_cust_max_age

71

In [199]:
#avarage age of defaulters 
def_avg_age = np.mean(def_cust.age)
def_avg_age

39.53496932515338

In [201]:
#median value of defaulters
def_median_age = np.median(def_cust.age)
def_median_age

38.0

In [205]:
#Standed deviation of defaulters 
def_std_age = np.std(def_cust.age)
def_std_age

9.352061114265064

In [247]:
bank11 = def_cust.describe().T
bank11

count          mean           std     min     25%      50%      75%  \
Cust_num  815.0  17491.235583  10447.241863     8.0  9011.5  16380.0  25168.5   
age       815.0     39.534969      9.357804    21.0    32.0     38.0     47.0   
balance   815.0   -137.624540    906.704678 -8019.0  -339.0     -7.0     15.0   
day       815.0     16.385276      8.564183     1.0     8.0     17.0     23.0   
duration  815.0    239.115337    214.512375     3.0    97.0    172.0    305.0   
campaign  815.0      3.148466      3.628145     1.0     1.0      2.0      4.0   
pdays     815.0     18.042945     73.651493    -1.0    -1.0     -1.0     -1.0   
previous  815.0      0.268712      1.629787     0.0     0.0      0.0      0.0   

              max  
Cust_num  44607.0  
age          71.0  
balance   16486.0  
day          31.0  
duration   1558.0  
campaign     36.0  
pdays       521.0  
previous     25.0

# Age:
in above table total number of defaulters is 815
defaulters average is 40 
deviation of age is 9
min defaulter customer age is 21
first 25% customers defaulters max page is 32
first 50% default customer max age is 38
75% default customer max age is 47

#balance:
defaulters customer average balance is -138
defaulters stranded deviation is 907
min defaulters customer balance is -8019
first 25% customers max balance is -339
50% customer max balance is -7
75% customers max balance is 15




In [270]:
#defaulters in marital type: 
def_married = bank[(bank["default"]=="yes") & (bank["marital"]=="married")]
def_single = bank[(bank["default"]=="yes") & (bank["marital"]=="single")]
def_divorced = bank[(bank["default"]=="yes") & (bank["marital"]=="divorced")]
print("married peoples defaulters", len(def_married))
print("% of married peoples defaulters", (len(def_married)/len(bank11)))
print("defaulters in single's", len(def_single))
print("% of single defaulters", (len(def_single)/len(bank11)))
print("defaulters in divorced ", len(def_divorced))
print("% of defaulters defaulters", (len(def_divorced)/len(bank11)))

married peoples defaulters 448
% of married peoples defaulters 56.0
defaulters in single's 239
% of single defaulters 29.875
defaulters in divorced  128
% of defaulters defaulters 16.0


In [297]:
bank1= bank[bank["default"]=="yes"]
marial_def = pd.value_counts(bank1["marital"])/len(bank1)*100
marial_def

married     54.969325
single      29.325153
divorced    15.705521
Name: marital, dtype: float64